media-production/use-cases-and-gap-analysis-frame-accurate-sync.md

```diff
@@ -0,0 +1,43 @@
+# Use Cases and Gap Analysis: Frame Accuracy for Media Production
+
+This document collects use cases and requirements for new or changed Web APIs related to frame accuracy to improve support for media production.
+
+## Functional Gaps
+
+### The media timeline is not precise enough to identify individual frames
+
+[HTML] does not expose any precise mechanism to assess the time, from a user's wall clock perspective, at which a particular media frame is going to be rendered. A web application may only infer this information by looking at the [media element]'s [currentTime] property to infer the frame being rendered and the time at which the user will see the next frame. This has several limitations:
+
+* [currentTime] is represented as a double value, which does not allow it to identify individual frames due to rounding errors. This is a [known issue][html#109].
+* [currentTime] is updated at a user-agent defined rate (typically the rate at which [time marches on] runs), and is kept stable while scripts are running. When a web application reads [currentTime], it cannot tell when this property was last updated, and thus cannot reliably assess whether this property still represents the frame currently being rendered.
```

Comment:
I'm not clear whether the request is to update `currentTime` while scripts are running or pause media progress until scripts have completed?

Reply:
A third option would be to expose the timestamp (using the `Performance.now()` clock) at which `currentTime` was last updated. `Performance.now()` keeps running while scripts are running, so web apps could run the computation themselves. In any case, I read it more as a gap than as a request for a particular solution.